### PR TITLE
Feature/docker uptime

### PR DESCRIPTION
**Pull request by yadima**
Added netuitivedocker collector uptime feature.

### CODE REVIEW

File: src/collectors/netuitivedocker/netuitivedocker.py [hunk 1]
```diff
@@ -123,6 +124,40 @@ def print_minimal_metric(cc, name):
             if total != 0:
                 self.publish(name + '.netuitive.docker.cpu.container_cpu_percent', 100.0 * usage / total)
 
+        def print_uptime(cc):
```
Comment (TheConnMan) [inline src/collectors/netuitivedocker/netuitivedocker.py]: `print_uptime` is an odd name for a method which collects uptime. What about `collect_uptime`?

File: src/collectors/netuitivedocker/netuitivedocker.py [hunk 2]
```diff
@@ -123,6 +124,40 @@ def print_minimal_metric(cc, name):
...
+            """
+            Publishes containers' uptime
+            """
+            for i in cc.containers():
```
Comment (TheConnMan) [inline src/collectors/netuitivedocker/netuitivedocker.py]: Collecting all containers again for every collection my adversely affect collection time. Why not perform this logic inside the existing container collection loop?

File: src/collectors/netuitivedocker/netuitivedocker.py [hunk 3]
```diff
@@ -123,6 +124,40 @@ def print_minimal_metric(cc, name):
...
+                name = i['Names'][0][1:]
+                try:
+                    if 'About' in i['Status']:
+                        status = 1
+                    else:
+                        status = [int(s) for s in i['Status'].split() if s.isdigit()][0]	
+
+                    uptime_minutes = 0
+                    if 'year' in i['Status']:
```
Comment (TheConnMan) [inline src/collectors/netuitivedocker/netuitivedocker.py]: This is an inaccurate representation of how long the container has been up. You'll only ever get the largest unit of how long the container has been up.

I recommend using the `Created` property on `i` which is the epoch time that the container was created. Subtract it from the current time to get an uptime accurate to the second.

**Review by TheConnMan — CHANGES_REQUESTED** [review]

File: src/collectors/netuitivedocker/netuitivedocker.py (hunk 3 above)
Comment (Grumblesmith) [reply src/collectors/netuitivedocker/netuitivedocker.py]: Is that property updated if the container is stopped then started?

Comment (TheConnMan) [reply src/collectors/netuitivedocker/netuitivedocker.py]: Good point, probably not. Even better is `cc.inspect_container(i['Id'])['State']['StartedAt']`.

File: src/collectors/netuitivedocker/netuitivedocker.py (hunk 1 above)
Comment (yadima) [reply src/collectors/netuitivedocker/netuitivedocker.py]: Done. Changed the name to collect_uptime in the last commit :)

File: src/collectors/netuitivedocker/netuitivedocker.py (hunk 2 above)
Comment (yadima) [reply src/collectors/netuitivedocker/netuitivedocker.py]: Done changes in the last commit. Moved the collect_uptime call inside the exisiting loop at the bottom.

File: src/collectors/netuitivedocker/netuitivedocker.py (hunk 3 above)
Comment (yadima) [reply src/collectors/netuitivedocker/netuitivedocker.py]: Done changes in the last commit. Rewritten the code with cc.inspect_container method.

**Review by TheConnMan — APPROVED** [review]

**Review by Grumblesmith — APPROVED** [review]

**Pull request merged**